.github/workflows/wheels.yaml

```diff
@@ -90,3 +90,31 @@ jobs:
         uses: pypa/gh-action-pypi-publish@release/v1
         with:
           packages-dir: ./wheelhouse/
+
+  publish-testpypi:
+    needs: wheels
+    name: Publish wheels to PyPI
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/testpypi')
+
+    environment:
+      name: testpypi
+      url: https://pypi.org/p/segyio
+
+    permissions:
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: wheel-*
+          merge-multiple: true
+          path: ./wheelhouse/
+
+      - name: Publish wheels to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: ./wheelhouse/
+          repository-url: https://test.pypi.org/legacy/
```

Comment:
This should be almost a 1 to 1 copy of our intended workflow for publishing to PyPI to verify that the workflow makes sense. Only things needed to publish to TestPyPI were changed.